detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/Project/ProjectReferenceResolver.cs

```diff
@@ -15,12 +15,23 @@ class ProjectReferenceResolver : DependencyResolver
 
         private string ProjectPath;
         private NugetSearchService NugetSearchService;
+        private HashSet<PackageId> Packages;
+        private bool CheckVersionOverride;
+        
         public ProjectReferenceResolver(string projectPath, NugetSearchService nugetSearchService)
         {
             ProjectPath = projectPath;
             NugetSearchService = nugetSearchService;
         }
-
+        
+        public ProjectReferenceResolver(string projectPath, NugetSearchService nugetSearchService, HashSet<PackageId> packages, bool checkVersionOverride)
```

Comment:
Add another constructor to extend support and pass along Package Versions which are parsed from Directory.Packages.props file. This will ensure we differentiate between Nuget Project Inspector and Nuget Solution Inspector.